### ISSUE TITLE
Tapping on bus marker opens stop information

### ISSUE THREAD

**Issue by bbodenmiller** [OneBusAway/onebusaway-android#428]
When on the show route view, when I try to click the bus to get real-time status information it normally ends up clicking the stop rather than the bus. To address this, I suggest hiding the stops at a certain zoom level where it no longer makes sense that you'd be trying to click an individual stop.

**Comment by bbodenmiller** [comment]
For example, here I tried to click the bus icon but ended up with the stop selected:

![screenshot_20160229-082459](https://cloud.githubusercontent.com/assets/1192780/13400736/fc2b73f4-debd-11e5-93d5-0b82117765d8.png)

**Comment by barbeau** [comment]
@bbodenmiller Yes, I've noticed this too - it mainly happens with north/south routes as shown, where the bus marker overlaps and underlying stop.  A workaround is to rotate the map using two fingers so that the north/south route runs left to right (instead of top to bottom) - the bus markers don't rotate with the map (arguably another issue), so that will move the bus marker away from the stop.  In v2.0.8 I added this workaround info in the tutorial.

I personally consider this a bug in Google's Android Maps v2 marker overlay handling - IIRC this same problem doesn't happen on the Amazon Fire Phone, as Amazon's implementation correctly passes the click event to the bus marker, not the stop marker.  In Android Maps v2 there also isn't a concept of z-order for markers, so we can't manually force one on top of the other to intercept click events.

As you mention, another workaround is to hide stops or make them untappable at a certain zoom level.  This is certainly possible, although I think there needs to be some careful thought about how exactly this is done.

**Comment by barbeau** [comment]
The June 27th release of Android Maps API v2 now supports a z-index, so we can try to resolve this issue using the new z-index setting:
https://developers.google.com/maps/documentation/android-api/releases

**Comment by barbeau** [comment]
Blocked by issue updating Google Play Services to 9.2.0 - https://github.com/OneBusAway/onebusaway-android/pull/577.

**Comment by barbeau** [comment]
#577 is now resolved and merged, so this issue is unblocked.

**Comment by barbeau** [comment]
Initial work towards this is in https://github.com/CUTR-at-USF/onebusaway-android/tree/zIndex branch.
